core-primitives/stf-executor/src/executor.rs

```diff
@@ -232,6 +231,16 @@ where
 			.map(into_map)?
 			.into();
 
+		// Update parentchain block on all states.
+		let state_handler = Arc::new(GlobalFileStateHandler);
```

Comment:
Why are we using the `GlobalFileStateHandler` here? It should only be used on top-level in the enclave. If we need a state handler here, we take it as a member in the constructor (and abstracted), which is already the case if I see correctly. So use the `self.state_handler` instead.